### PR TITLE
CIRC-1467: TLR confirmation notice, two fields don't populate

### PR DESCRIPTION
**Pull request by yusuf-murodov-epam**
## Purpose
[CIRC-1467](https://issues.folio.org/browse/CIRC-1467)
_TLR confirmation notice: Two fields don't populate_

### CODE REVIEW

File: src/main/java/org/folio/circulation/infrastructure/storage/inventory/LocationRepository.java [hunk 1]
```diff
@@ -105,7 +105,28 @@ public CompletableFuture<Result<MultipleRecords<Location>>> fetchLocations(
...
+    return multipleRecordsResult.combineAfter(
+      locations -> getInstitutions(locations.getRecords()), (locations, institutions) ->
+        locations.mapRecords(location -> location.withInstitutionRepresentation(
+          institutions.getOrDefault(location.getInstitutionId(),null))));
+  }
+
+  private CompletableFuture<Result<MultipleRecords<Location>>> loadCampusesForLocations(
+    Result<MultipleRecords<Location>> multipleRecordsResult) {
+
+
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/infrastructure/storage/inventory/LocationRepository.java:125]: redundant empty line

File: src/main/java/org/folio/circulation/infrastructure/storage/inventory/LocationRepository.java [hunk 2]
```diff
@@ -105,7 +105,28 @@ public CompletableFuture<Result<MultipleRecords<Location>>> fetchLocations(
...
+      .thenCompose(this::loadInstitutionsForLocations);
+  }
+
+  private CompletableFuture<Result<MultipleRecords<Location>>> loadInstitutionsForLocations(
+    Result<MultipleRecords<Location>> multipleRecordsResult) {
+
+    return multipleRecordsResult.combineAfter(
+      locations -> getInstitutions(locations.getRecords()), (locations, institutions) ->
+        locations.mapRecords(location -> location.withInstitutionRepresentation(
+          institutions.getOrDefault(location.getInstitutionId(),null))));
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/infrastructure/storage/inventory/LocationRepository.java:119]:
```suggestion
          institutions.getOrDefault(location.getInstitutionId(), null))));
```

File: src/test/java/api/requests/scenarios/PageRequestWorkflowTests.java [hunk 3]
```diff
@@ -84,4 +91,32 @@ void itemCannotBeCheckedOutToOtherPatronWhenItemIsPagedAndNotYetBeingFulfilled()
 
     assertThat(pagedSmallAngryPlanet, hasItemStatus(PAGED));
   }
+
+  @Test
+  void titleLevelRequestConfirmationNoticeWithValidLocationShouldBeSentWithEnabledTlr() {
+
```
Comment (roman-barannyk) [inline src/test/java/api/requests/scenarios/PageRequestWorkflowTests.java:97]: redundant empty line

Comment (yusuf-murodov-epam) [reply src/test/java/api/requests/scenarios/PageRequestWorkflowTests.java]: fixed

File: src/main/java/org/folio/circulation/infrastructure/storage/inventory/LocationRepository.java (hunk 2 above)
Comment (yusuf-murodov-epam) [reply src/main/java/org/folio/circulation/infrastructure/storage/inventory/LocationRepository.java]: fixed

File: src/main/java/org/folio/circulation/infrastructure/storage/inventory/LocationRepository.java (hunk 1 above)
Comment (yusuf-murodov-epam) [reply src/main/java/org/folio/circulation/infrastructure/storage/inventory/LocationRepository.java]: fixed

**Review by roman-barannyk — APPROVED** [review]

**Review by sherzodn — APPROVED** [review]

File: src/main/java/org/folio/circulation/resources/TitleLevelRequestNoticeSender.java [hunk 4]
```diff
@@ -34,8 +34,14 @@ public Result<RequestAndRelatedRecords> sendNoticeOnRequestCreated(
     if (tlrSettings.isTitleLevelRequestsFeatureEnabled()
       && tlrSettings.getConfirmationPatronNoticeTemplateId() != null) {
 
-      PatronNoticeEvent requestCreatedEvent = createPatronNoticeEvent(request, eventType);
-      applyTlrConfirmationNotice(request.getTlrSettingsConfiguration(), requestCreatedEvent);
+      locationRepository.loadCampus(request.getItem().getLocation())
+        .thenCompose(r -> r.after(locationRepository::loadInstitution))
+        .thenApply(r -> r.map(request.getItem()::withLocation))
+        .thenApply(r -> r.map(request::withItem))
+        .thenAccept(r -> r.after(result -> {
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/resources/TitleLevelRequestNoticeSender.java:41]: which type is `result`? Maybe we can give it a more descriptive name?

File: src/main/java/org/folio/circulation/resources/TitleLevelRequestNoticeSender.java [hunk 5]
```diff
@@ -34,8 +34,14 @@ public Result<RequestAndRelatedRecords> sendNoticeOnRequestCreated(
...
+          PatronNoticeEvent requestCreatedEvent = createPatronNoticeEvent(result, eventType);
+          return applyTlrConfirmationNotice(result.getTlrSettingsConfiguration(), requestCreatedEvent);
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/resources/TitleLevelRequestNoticeSender.java:43]: Can we extract method here?

**Review by alexanderkurash — APPROVED** [review]

File: src/main/java/org/folio/circulation/resources/TitleLevelRequestNoticeSender.java (hunk 5 above)
Comment (yusuf-murodov-epam) [reply src/main/java/org/folio/circulation/resources/TitleLevelRequestNoticeSender.java]: fixed

File: src/main/java/org/folio/circulation/resources/TitleLevelRequestNoticeSender.java (hunk 4 above)
Comment (yusuf-murodov-epam) [reply src/main/java/org/folio/circulation/resources/TitleLevelRequestNoticeSender.java]: fixed

**Pull request merged**